relentless/collections.py

```diff
@@ -0,0 +1,339 @@
+"""
+Collections
+===========
+
+This module contains some data structures used in :mod:`relentless` as an
+alternative to Python's general purpose containers, for ease of use in constructing
+potentials and performing computations during the optimization workflow.
```

Comment:
In the docs, this `:mod:` link is coming back to the developer page, not sure why. Suggesting this reword to avoid it:
```suggestion
This module contains some customized data structures used for constructing
potentials and performing computations during the optimization workflow.
```

(Edit: the link issue might get fixed once you fix the docstring per my comment above. In that case, you might not need to do anything here.)